Ingress/controllers/nginx-third-party/README.md

```diff
@@ -0,0 +1,268 @@
+```
+$ curl -v 172.17.4.99/foobar -H 'Host: foo.bar.com'
+*   Trying 172.17.4.99...
+* Connected to 172.17.4.99 (172.17.4.99) port 80 (#0)
+> GET /foobar HTTP/1.1
+> Host: foo.bar.com
+> User-Agent: curl/7.43.0
+> Accept: */*
+>
+< HTTP/1.1 404 Not Found
+< Server: nginx/1.9.8
+< Date: Tue, 15 Dec 2015 13:48:18 GMT
+< Content-Type: text/html
+< Transfer-Encoding: chunked
+< Connection: keep-alive
+< Vary: Accept-Encoding
+<
+default backend - 404
+* Connection #0 to host 172.17.4.99 left intact
+```
+
+(this test checked that the default backend is properly working)
+
+*Replacing the default backend with a custom one we can change the default error pages provided by nginx*
+
+# Exposing TCP services
+
+First we need to remove the running
+```
+kubectl delete rc nginx-ingress-3rdpartycfg
+```
+
+```
+kubectl create -f examples/rc-tcp.yaml
+```
+
+Now we add the annotation to the replication controller that indicates with services should be exposed as TCP:
+The annotation key is `nginx-ingress.kubernetes.io/tcpservices`. You can expose more than one service using comma as separator.
+Each service must contain the namespace, service name and port to be use as public port
+
+```
+kubectl annotate rc nginx-ingress-3rdpartycfg "nginx-ingress.kubernetes.io/tcpservices=default/echoheaders-x:9000"
+```
+
+*Note:* the only reason to remove and create a new rc is that we cannot open new ports dynamically once the pod is running.
+
+
+Once we run the `kubectl annotate` command nginx will reload.
+
+Now we can test the new service:
+```
+$ (sleep 1; echo "GET / HTTP/1.1"; echo "Host: 172.17.4.99:9000"; echo;echo;sleep 2) | telnet 172.17.4.99 9000
+
+Trying 172.17.4.99...
+Connected to 172.17.4.99.
+Escape character is '^]'.
+HTTP/1.1 200 OK
+Server: nginx/1.9.7
+Date: Tue, 15 Dec 2015 14:46:28 GMT
+Content-Type: text/plain
+Transfer-Encoding: chunked
+Connection: keep-alive
+
+f
+CLIENT VALUES:
+
+1a
+client_address=10.2.84.45
+
+c
+command=GET
+
+c
+real path=/
+
+a
+query=nil
+
+14
+request_version=1.1
+
+25
+request_uri=http://172.17.4.99:8080/
+
+1
+
+
+f
+SERVER VALUES:
+
+28
+server_version=nginx: 1.9.7 - lua: 9019
+
+1
+
+
+12
+HEADERS RECEIVED:
+
+16
+host=172.17.4.99:9000
+
+6
+BODY:
+
+14
+-no body in request-
+0
+```
+
+# SSL
+
+Currently Ingress rules does not contains SSL definitions. In order to support SSL in nginx this controller uses secrets mounted inside the directory `/etc/nginx-ssl` to detect if some Ingress rule contains a host for which it is possible the creation of an SSL server.
+
+`SECRET_NAME=secret-echoheaders-1 HOSTS=foo.bar.com ./examples/certs.sh`
+
+```
+kubectl create -f secret-secret-echoheaders-1-foo.bar.com.json
+```
+
+```
+$ kubectl get secrets
+NAME                   TYPE                                  DATA      AGE
+secret-echoheaders-1   Opaque                                2         9m
+
+```
+
+Like before we need to remove the running nginx rc
+```
+kubectl delete rc nginx-ingress-3rdpartycfg
+```
+
```

Comment:
please explain what you're doing, i.e something like: create a secret with a wildcard cert (or SAN) for all hosts in the ingress, then re-create the RC with this secret mounted under /etc/nginx-ssl. Now the nginx controller is capable of terminating ssl on 443 using the given secret.